Plane: allow reverse thrust in non auto-throttle modes if USE_REVERSE_THRUST=0

shellixyz opened the pull request:
Right now it is not possible to use reverse thrust with MANUAL/FBWA if USE_REVERSE_THRUST is set to 0

Hwurzburg commented:
Manual works...other non auto throttle modes do not

shellixyz commented:
Could this be merged ? Seems like a straightforward fix and I didn't see any issue while testing

rmackay9 commented:
I think the commit should have "Plane:" prefixed to fit with the normal conventions.

Any chance you could attend next week's dev call?  It's on Tuesday at 7am Canberra time.  I'm not a plane expert but if it's discussed on the call (which it will be now that I've added the DevCallTopic) it would be good to have you there to talk about it with the Plane devs.

shellixyz commented:
> I think the commit should have "Plane:" prefixed to fit with the normal conventions.

I added the prefix :+1: 

> Any chance you could attend next week's dev call? It's on Tuesday at 7am Canberra time. I'm not a plane expert but if it's discussed on the call (which it will be now that I've added the DevCallTopic) it would be good to have you there to talk about it with the Plane devs.

I've set a reminder, I will try to attend :)

shellixyz commented:
Sorry I'm not going to be able to attend the dev call today, it is just too late. I will be present for the EU dev call

tridge commented on ArduPlane/reverse_thrust.cpp:
```diff
@@ -25,7 +25,7 @@ bool Plane::allow_reverse_thrust(void) const
     bool allow = false;
 
-    if (g.use_reverse_thrust == USE_REVERSE_THRUST_NEVER || !have_reverse_thrust()) {
+    if (!have_reverse_thrust()) {
```
maybe instead a single bit in USE_REV_THRUST for manual throttle modes?

magicrub replied:
agreed. Better to add a new entry like "USE_REVERSE_THRUST_non_auto_throttle_modes" but with a better name

shellixyz replied:
Why would you want to disable reverse thrust in manual modes and either lose half of the throttle stick or have to lose a channel to assign to rev thrust to not lose half of the throttle stick ?

Anyway that's a fix for a bug not a pull request for changing the semantic of a param or adding a new one. Feel free to merge as is... or not

Hwurzburg replied:
it does not mean you lose half the throttle stick...in plane, often the reverse thrust region is very small at the bottom of the stick to help in steepening approaches....but, you may not want that in FBWA or QSTABILIZE.....MANUAL should always be enabled....but having a bit that ignores the reverse end of the stick for manual pilot controlled stabilize modes, [or even better, overriding THR_MIN to zero in those modes (giving full throttle stick control for the forward range)...my thought] , is what was discussed in the DEV CALL..rather than this....scope increase of course, but the consensus of what a fix should be....

shellixyz replied:
Overriding THR_MIN would make the most sense in my opinion.

The thing is that this param is broken right now and besides what could be done to improve things this PR doesn't change the intended behavior so independently of what should be done for manual modes it would be a good idea to at least merge it while a concensus on what should be done to allow to disable rev thrust for manual modes is found.

Hwurzburg replied:
the parameter was discussed as being intended, and applicable to, auto-throttle modes only...the description should be corrected actually until bit or bits are added to allow reverse thrust in FBWA,ACRO,STABILIZE individually or as a composite...changing behavior now could cause user issues to those already using reverse thrust in AUTO modes but not expecting it in the above modes is what the decision was....

shellixyz replied:
The only thing this PR is changing is the fact that it was disabling it in manual modes if set to 0 which didn't make any sense. 

So "could cause user issues to those already using reverse thrust in AUTO modes but not expecting it in the above modes" doesn't make any sense in the context of how it has always been working besides the fact that disabling rev thrust in all auto modes also was disabling it in manual modes which again didn't make any sense
